opentelemetry-api/src/logs/record.rs

```diff
@@ -4,6 +4,22 @@ use crate::{
 };
 use std::{borrow::Cow, time::SystemTime};
 
+#[derive(Debug, Clone)]
+/// Timestamp for when the record was observed by OpenTelemetry
+/// Defaults to the current time.
+pub struct ObservedTimestamp {
+    /// Time when the record was observed by OpenTelemetry
+    pub time: SystemTime,
+}
```

Comment:
Hmm do we expect to add more fields in this struct? If we do we should add `#[non_exhaustive]` attribute for this struct. If we don't why not just use `SystemTime` in the LogRecord and have the default there?

Reply:
Thanks, @TommyCpp. How can I add the default for `observed_timestamp` within LogRecord? This was what I was originally trying out, but seems not doable without explicitly defining the default trait for LogRecord.

Reply:
Ok. I have added the default trait for LogRecord. This was much simpler as rest of the field are optional, and so default to `None`. Thanks for the suggestion.